python/lsst/pipe/base/pipeline.py

```diff
@@ -136,7 +136,10 @@ def __str__(self):
     def __eq__(self, other: object) -> bool:
         if not isinstance(other, TaskDef):
             return False
-        return self.config == other.config and\
+        configDict = self.config.toDict()
+        otherDict = other.config.toDict()
+        configEq = configDict == otherDict or str(configDict) == str(otherDict)
```

Comment:
Following up slack conversation - I think `Config.__eq__` should already handle `nan` comparison adequately: https://github.com/lsst/pex_config/blob/master/python/lsst/pex/config/config.py#L1368-L1379. There is also [`Config.compare()`](https://github.com/lsst/pex_config/blob/master/python/lsst/pex/config/config.py#L1393) method which does precision-base comparison but in this case I think we want bitwise comparison instead. If `__eq__` is not working then I agree that it needs to be fixed in`pex_config`.

Reply:
Somewhere in the hierarchy `__eq__` was not working right because it was attempting to look at the id of the two objects, even though their contents were the same. I will need to try and recreate that, the details are a bit fuzzy in my mind.